.github/workflows/cli.yml

```diff
@@ -10,28 +10,23 @@ jobs:
     env:
       CARGO_INCREMENTAL: 0
     steps:
-      - name: (PR comment) Get PR branch
-        if: ${{ github.event_name == 'issue_comment' }}
-        uses: xt0rted/pull-request-comment-branch@v2
-        id: comment-branch
-
-      - name: (PR comment) Set latest commit status as pending
-        if: ${{ github.event_name == 'issue_comment' }}
+      - name: (PR review) Set latest commit status as pending
+        if: ${{ github.event_name == 'pull_request_review' }}
         uses: myrotvorets/set-commit-status-action@v2.0.1
         with:
-          sha: ${{ steps.comment-branch.outputs.head_sha }}
+          sha: ${{ github.events.review.commit_id }}
           token: ${{ secrets.GITHUB_TOKEN }}
           context: CLI - Linux
           status: pending
 
-      - name: (PR comment) Checkout PR branch
-        if: ${{ github.event_name == 'issue_comment' }}
+      - name: (PR review) Checkout review SHA
+        if: ${{ github.event_name == 'pull_request_review' }}
         uses: actions/checkout@v3
         with:
-          ref: ${{ steps.comment-branch.outputs.head_ref }}
+          ref: ${{ github.events.review.commit_id }}
 
       - name: Checkout branch
-        if: ${{ github.event_name != 'issue_comment' }}
+        if: ${{ github.event_name != 'pull_request_review' }}
```

Comment:
you could avoid this construct and create a more hard-linked reference: set an input to this reusable workflow to provide the `ref`, emit a "validated commit ID" in the `validate` job and pass it as input to these workflows.

the pattern is repeated in several reusable workflows.

Reply:
Agreed, I did notice this boilerplate repeated across workflows that could be DRY'd up, but didn't want to attempt that refactoring with this change.

Reply:
ok, sounds good to me